src/main/java/duke/taskmanagement/TaskManager.java

```diff
@@ -0,0 +1,186 @@
+package duke.taskmanagement;
+
+import duke.customexceptions.EmptyDescriptionException;
+import duke.customexceptions.EmptyTimingDetailsException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class TaskManager {
+    private int taskUniqueID = 0;
+    private static ArrayList<Task> taskList = new ArrayList<>();
+    private static TaskRecorder taskRecorder = new TaskRecorder();
+
+    public TaskManager() {
+        try {
+            ArrayList<String[]> fileData = taskRecorder.loadData();
+            for (String[] data : fileData) {
+                String input = data[1];
+                String command = data[1].split(" ")[0];
+                String description = getDescription(input);
+                switch (command) {
+                case "todo":
+                    taskList.add(new Todo(description, taskUniqueID));
+                    break;
+                case "deadline":
+                    String by = getTimingDetails(input);
+                    taskList.add(new Deadline(description, taskUniqueID, by));
+                    break;
+                case "event":
+                    String at = getTimingDetails(input);
+                    taskList.add(new Event(description, taskUniqueID, at));
+                    break;
+                }
+                if (data[0].equals("1")) {
+                    taskList.get(taskList.size() - 1).setIsMarked();
+                }
+                taskUniqueID++;
+            }
+        } catch (IOException e) {
+            System.out.println("Oops! Something went wrong while loading the data!");
+        } catch (EmptyDescriptionException e) {
+            System.out.println("Oops! Please type in a description!");
+        } catch (EmptyTimingDetailsException e) {
+            System.out.println("Oops! Please indicate the time for the deadline you want to add!");
+        }
+    }
+
+    public void listTasks() {
+        if (taskList.size() > 0) {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i < taskList.size(); i++) {
+                System.out.println(i + 1 + "." + taskList.get(i).toString());
+            }
+        } else {
+            System.out.println("Wow, such empty");
+        }
+    }
+
+    public void addTodo(String userInput) {
+        try {
+            String description = getDescription(userInput);
+            taskList.add(new Todo(description, taskUniqueID));
+            printMessageForAdding(taskList.get(taskList.size() - 1));
```

Comment:
There are 3 methods invoked on this line, to extract the newly added task. How about instantiating the toDo task in a different line? Then you could reuse the variable here.

Reply:
Hi stephanie,

Do you mean to add:
`newTodo = new Todo(description, taskUniqueID);
taskList.add(newTodo);`